packages/@react-spectrum/textfield/src/TextFieldBase.tsx

```diff
@@ -54,13 +54,17 @@ function TextFieldBase(props: TextFieldBaseProps, ref: Ref<TextFieldRef>) {
     inputRef,
     isLoading,
     loadingIndicator,
+    value,
+    defaultValue,
```

Comment:
Let's go ahead and add a `readOnlyText` prop here instead. As it is now, this is affecting other components like ColorField, ComboBox, NumberField, etc. See the ColorField isReadOnly story

Move `let inputValue = value ?? defaultValue ?? ''` up into TextField and TextArea as well.